Remove fallback to default device open

quic-lxu5 opened the pull request:
In case of default domain ID, the device open request falls back to DEFAUL_DEVICE which will be defined in the libs to which the application is linked. Remove this legacy fallback logic to avoid any confusion and unexpected observations.

quic-ekangupt commented:
Can you also clean-up the following:

1. This function: https://github.com/qualcomm/fastrpc/blob/8c8e7e050a4bde06ea75623115f2c6e108b1a323/src/adsp_default_listener.c#L95
2. DEFAULT_DEVICE from here:
https://github.com/qualcomm/fastrpc/blob/8c8e7e050a4bde06ea75623115f2c6e108b1a323/src/fastrpc_ioctl.c#L44
https://github.com/qualcomm/fastrpc/blob/8c8e7e050a4bde06ea75623115f2c6e108b1a323/inc/fastrpc_ioctl.h#L45
https://github.com/qualcomm/fastrpc/blob/8c8e7e050a4bde06ea75623115f2c6e108b1a323/src/fastrpc_apps_user.c#L3264

quic-lxu5 committed:
Remove fallback to default device open

In case of default domain ID, the device open request falls back to
DEFAUL_DEVICE which will be defined in the libs to which the application
is linked. Remove this legacy fallback logic to avoid any confusion
and unexpected observations.

Signed-off-by: Ling Xu <ling.xu@oss.qualcomm.com>
